services/src/main/java/org/keycloak/services/DefaultKeycloakSessionFactory.java

```diff
@@ -229,6 +217,10 @@ private boolean isEnabled(ProviderFactory factory, Config.Scope scope) {
         if (factory instanceof EnvironmentDependentProviderFactory) {
             return ((EnvironmentDependentProviderFactory) factory).isSupported();
         }
+        if (factory instanceof FeatureDependentProviderFactory) {
+            return Optional.ofNullable(((FeatureDependentProviderFactory) factory).getRequiredFeatures()).orElse(Collections.emptySet())
+                    .stream().allMatch(requiredFeature -> Profile.isFeatureEnabled(requiredFeature));
+        }
```

Comment:
@stianst @mstruk Done.  Just a though - 'EnvironmentDepndent' becomes a bit of a misnomer here.  Perhaps something like 'ConditionalProviderFactory' going forward?  Two hard things.